contracts/governance/proposals.cairo

```diff
@@ -0,0 +1,287 @@
+    if (block_cmp == 1){
+        return (res=0);
+    }
+    
+    let (gov_token_addr) = governance_token_address.read();
+    let (nay_tally) = proposal_total_yay.read(prop_id);
+    let (yay_tally) = proposal_total_nay.read(prop_id);
+    let total_tally = yay_tally + nay_tally;
+    let total_tally_uint256 = intToUint256(total_tally);
+    
+
+    with_attr error_message("unable to check quorum"){
+        let share = Uint256(low = QUORUM, high = 0);
+        let (tally_multiplied, carry) = uint256_mul(total_tally_uint256, share);
+        assert carry.low = 0;
+        assert carry.high = 0;
+        let (total_eligible_votes) = IERC20.totalSupply(contract_address=gov_token_addr);
+    }
+
+    let (cmp_res) = uint256_lt(total_eligible_votes, tally_multiplied);
+    if (cmp_res == 0) {
+        return (res=-1); // didn't meet quorum
+    }
+
+    let yay_or_nay = yay_tally - nay_tally;
+    if (yay_or_nay == 0){
+        return (res=-1); // yay_tally = nay_tally
+    }
+    
+    let nn = is_nn(yay_or_nay);
+    if (nn == 1) {
+        return (res=1); // yay_tally > nay_tally
+    }else{
+        return (res=-1); // yay_tally < nay_tally
+    }
+}
+
+
+// @notice Investors don't hold tokens as of launch. They can vote only from whitelisted addresses.
+// When investor votes, the real voting power of all investors is equivalent to totalSupply - balanceOf(TEAM_MULTISIG_ADDR).
+// The real voting power is calculated in this function. Voting power of community : team : investors is 2:1:1.
+// @dev TEAM_MULTISIG_ADDR holds the team's tokens. 
+@external
+func vote_investor{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    prop_id: felt,
+    opinion: felt
+) {
+    alloc_locals;
+
+    // Checks
+    with_attr error_message("opinion must be either 1 = yay or -1 = nay"){
+        assert_not_zero(opinion);
+        assert_le(opinion, 1);
+        assert_le(-1, opinion);
+    }
+
+    let (caller_addr) = get_caller_address();
+    let (investor_voting_power_l) = investor_voting_power.read(caller_addr);
+    local investor_voting_power_local = investor_voting_power_l;
+    with_attr error_message("caller not a whitelisted investor"){
+        assert_not_zero(investor_voting_power_local);
+    }
+
+    let (curr_votestatus) = proposal_voted_by.read(prop_id, caller_addr);
+    with_attr error_message("already voted"){
+        assert curr_votestatus = 0;
+    }
+
+    let (end_block_number) = proposal_vote_ends.read(prop_id);
+    let (curr_block_number) = get_block_number();
+    with_attr error_message("voting already concluded"){
+        let block_diff = curr_block_number - end_block_number;
+        assert_not_zero(block_diff);
+        assert_nn(block_diff);
+    }
+
+    // Calculate real voting power
+    let (GOV_TOKEN_ADDRESS) = governance_token_address.read();
+    let (team_balance) = IERC20.balanceOf(contract_address=GOV_TOKEN_ADDRESS, account=TEAM_MULTISIG_ADDR);
```

Comment:
If I understand correctly, this assumes the team will have the tokens in a single account that requires multisig???

in reality the tokens that will be released will be stored on our wallets/accounts and those not yet released to us (either not yet minted or not yet "freed" will not be allowed to have any power)